src/modules/select/classes/select-base.ts

```diff
@@ -331,6 +341,9 @@ export abstract class SuiSelectBase<T, U> implements AfterContentInit, OnDestroy
     public abstract selectOption(option:T):void;
 
     protected findOption(options:T[], value:U):T | undefined {
+        if (this._compareWith) {
```

Comment:
I think rather than checking to see if the function has been assigned, it would be cleaner to initialise this property in the constructor with `(a, b) => a === b` , then you don't need this check.